src/backend/base/langflow/components/agents/LangSmithAgent.py

```diff
@@ -0,0 +1,49 @@
+from langchain import hub
+from langchain.agents import create_tool_calling_agent
+from langchain_core.prompts import ChatPromptTemplate, PromptTemplate, HumanMessagePromptTemplate
+from langflow.base.agents.agent import LCToolsAgentComponent
+from langflow.inputs import MultilineInput, StrInput, SecretStrInput
+from langflow.inputs.inputs import HandleInput
+
+
+class LangSmithAgentComponent(LCToolsAgentComponent):
+    display_name: str = "LangSmith Agent"
+    description: str = "Agent that uses LangSmith prompts"
+    icon = "LangChain"
+    beta = True
+    name = "LangSmithAgent"
+
+    inputs = LCToolsAgentComponent._base_inputs + [
+        HandleInput(name="llm", display_name="Language Model", input_types=["LanguageModel"], required=True),
+        MultilineInput(
+            name="system_prompt",
+            display_name="System Prompt",
+            info="System prompt for the agent.",
+            value="You are a helpful assistant.",
+        ),
```

Comment:
There is a conceptual problem here, it is that in LangSmith PromptHub we can have both a system prompt and a placeholder with human text entry

Reply:
Thanks for the comment @zee229 - can you clarify a bit? My thought was that this system prompt would essentially be the beginning of the full prompt - so, a user can sort of "augment" the existing langsmith prompt hub prompt with some additional information. But i haven't used prompt hub much, so I might be misunderstanding it. Can you suggest how this should be changed?

Reply:
> Thanks for the comment @zee229 - can you clarify a bit? My thought was that this system prompt would essentially be the beginning of the full prompt - so, a user can sort of "augment" the existing langsmith prompt hub prompt with some additional information. But i haven't used prompt hub much, so I might be misunderstanding it. Can you suggest how this should be changed?

https://github.com/langflow-ai/langflow/issues/2822#issuecomment-2253796080